app.spec.js

```diff
@@ -0,0 +1,85 @@
+var app = require('./app');
+var shell = require('shelljs');
+var request = require('supertest');
+
+describe('test the root path', () => {
+  test('it should respond to the get method', () => {
+    return request(app).get('/').then(response => {
+      expect(response.statusCode).toBe(200)
+    });
+  });
+});
+
+describe('user can make post request', () => {
+  beforeAll(() => {
+    shell.exec('npx sequelize db:create');
+  });
+  beforeEach(() => {
+    shell.exec('npx sequelize db:migrate');
+    shell.exec('npx sequelize db:seed:all');
+  });
+  afterEach(() => {
+    shell.exec('npx sequelize db:migrate:undo:all');
+  });
+
+  test('to create account and get an api key back', () => {
+    var body = {"email": "earl@example.com",
+                "password": "password",
+                "password_confirmation": "password"
+                }
+      return request(app).post('/api/v1/users')
+                       .type('form')
+                       .set("Content-Type", "application/json")
+                       .send(body)
+                       // alternative way of passing body data:
+                       // .send({"email": "earl@example.com",
+                       //       "password": "password",
+                       //       "password_confirmation": "password"})
+                       .then(response => {
+      expect(response.statusCode).toBe(201);
+      expect(Object.keys(response.body)).toContain('api_key')
+    });
+  });
+
+  test('sad path create account and get an api key back', () => {
+    var body = {"email": "earl@example.com",
+                "password": "password",
+                "password_confirmation": "pw"
+                }
+      return request(app).post('/api/v1/users')
+                       .type('form')
+                       .set("Content-Type", "application/json")
+                       .send(body)
+                       .then(response => {
+      expect(response.statusCode).toBe(401)
+    });
+  });
+
+  test("user can get a city's forecast", () => {
+    var body = {"api_key": "1234567890abcdef"}
+      return request(app).get('/api/v1/forecast?location=miami,fl')
+        .type('form')
+        .set("Content-Type", "application/json")
+        .set("Accept", "application/json")
+        .send(body)
+        .then(response => {
+          expect(response.statusCode).toBe(200);
+          expect(Object.keys(response.body["hourly"]["data"][0])).toContain("humidity");
```

Comment:
Nice. This part tripped me up until I found Object.keys() since JS doesn't let you call `.keys` on something directly. Hopefully it gave you less trouble!